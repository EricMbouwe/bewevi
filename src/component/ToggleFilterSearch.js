import React from 'react';
import PropTypes from 'prop-types';
import Logo from './Logo';

const ToggleFilterSearch = ({ handleClick, shownContainer }) => (
  <div className="toggle-filter-search main">
    <Logo />
    {shownContainer}
    <button type="button" onClick={handleClick} className="toggle-filter-search toggle-button">
      Filter
    </button>

  </div>
);

ToggleFilterSearch.propTypes = {
  handleClick: PropTypes.func.isRequired,
  shownContainer: PropTypes.shape().isRequired,
};

export default ToggleFilterSearch;
