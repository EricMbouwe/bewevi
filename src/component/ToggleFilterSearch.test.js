import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import ToggleFilterSearch from './ToggleFilterSearch';

let handleClick;
let shownContainer;
let renderReadyComponent;

jest.mock('./Logo');

beforeEach(() => {
  handleClick = jest.fn();
  shownContainer = <div>Shown Container</div>;
  renderReadyComponent = (
    <ToggleFilterSearch
      handleClick={handleClick}
      shownContainer={shownContainer}
    />
  );
});

afterEach(() => {
  handleClick.mockClear();
});

describe('<ToggleFilterSearch />', () => {
  it('renders given component as shownComponent and ', () => {
    render(
      renderReadyComponent,
    );

    expect(screen.getByText(/Shown Container/i)).toBeInTheDocument();
    // expect(screen.getByText(/bar/i)).toBeInTheDocument();
    // userEvent.click(screen.getByText(/bar/i));
    // expect(screen.getByText(/One Plant Page/i)).toBeInTheDocument();
  });

  it('renders Logo component ', () => {
    render(
      renderReadyComponent,
    );

    expect(screen.getByText(/This is a Logo/i)).toBeInTheDocument();
  });

  it('is triggering handleClick when the link is clicked', () => {
    render(
      renderReadyComponent,
    );

    userEvent.click(screen.getByText(/Filter/i));
    expect(handleClick.mock.calls.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      renderReadyComponent,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
