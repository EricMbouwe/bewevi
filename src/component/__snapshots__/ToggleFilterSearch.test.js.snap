// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleFilterSearch /> renders correctly 1`] = `
<div
  className="toggle-filter-search main"
>
  This is a Logo
  <div>
    Shown Container
  </div>
  <button
    className="toggle-filter-search toggle-button"
    onClick={[MockFunction]}
    type="button"
  >
    Filter
  </button>
</div>
`;
